drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/wizard/pages/FactPatternConstraintsPageTest.java

```diff
@@ -0,0 +1,282 @@
+                                    ModelField.FIELD_CLASS_TYPE.REGULAR_CLASS,
+                                    ModelField.FIELD_ORIGIN.DECLARED,
+                                    FieldAccessorsAndMutators.BOTH,
+                                    DataType.TYPE_STRING ),
+                            new ModelField( "homeAddress",
+                                    Boolean.class.getName(),
+                                    ModelField.FIELD_CLASS_TYPE.REGULAR_CLASS,
+                                    ModelField.FIELD_ORIGIN.DECLARED,
+                                    FieldAccessorsAndMutators.BOTH,
+                                    DataType.TYPE_BOOLEAN ),
+                            new ModelField( "number",
+                                    Integer.class.getName(),
+                                    ModelField.FIELD_CLASS_TYPE.REGULAR_CLASS,
+                                    ModelField.FIELD_ORIGIN.DECLARED,
+                                    FieldAccessorsAndMutators.BOTH,
+                                    DataType.TYPE_NUMERIC_INTEGER ) };
+
+
+        doAnswer( new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                ((Callback<ModelField[]>) invocation.getArguments()[1]).callback( modelFields );
+                return null;
+            }
+        } ).when( oracle ).getFieldCompletions( anyString(), any( Callback.class ) );
+
+        doAnswer( new Answer<String>() {
+            @Override
+            public String answer(InvocationOnMock invocation) throws Throwable {
+                String fieldName = invocation.getArgumentAt(1, String.class);
+                for(ModelField field : modelFields) {
+                    if(field.getName().equals(fieldName)) {
+                        return field.getClassName();
+                    }
+                }
+                return null;
+            }
+        } ).when( oracle ).getFieldType( anyString(), anyString() );
+
+        doAnswer( new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                invocation.getArgumentAt( 2, Callback.class ).callback( new String[] { "==", "!=", "in", "not in" } );
+                return null;
+            }
+        } ).when( oracle ).getOperatorCompletions( anyString(), anyString(), any( Callback.class ) );
+
+        conditions = new ArrayList<ConditionCol52>();
+        conditions.add( mock( ConditionCol52.class ) );
+        when( pattern.getChildColumns() ).thenReturn( conditions );
+        when( secondPattern.getChildColumns() ).thenReturn( conditions );
+    }
+
+    @Test
+    public void testPrepareView() throws Exception {
+        page.prepareView();
+
+        verify( view ).setAvailablePatterns( Arrays.asList( pattern, secondPattern ) );
+    }
+
+    @Test
+    public void testIsComplete() throws Exception {
```

Comment:
Test ```testIsNotComplete()``` is very similar; perhaps to avoid code duplication have both tests call a shared private method that takes a ```boolean``` valid-state parameter; used by your mock set-up and verification?